packages/tlschema/src/createTLSchema.ts

```diff
@@ -79,7 +85,13 @@ export function createTLSchema({
 		firstVersion: rootShapeTypeMigrations.firstVersion,
 		migrators: rootShapeTypeMigrations.migrators,
 		subTypeKey: 'type',
-		subTypeMigrations: Object.fromEntries(allShapeDefs.map((def) => [def.type, def.migrations])),
+		subTypeMigrations: Object.fromEntries(
+			(
+				allShapeDefs.filter((def) => def.migrations) as (CustomShapeTypeInfo & {
+					migrations: Migrations
+				})[]
+			).map((def) => [def.type, def.migrations])
+		),
```

Comment:
I don't think it's safe to filter them out here, the migration code relies on the migrations metadata existing for all sub types. You could change this to return an 'empty' migrations object for shapes with no migrations defined.